src/libraries/System.Net.Quic/tests/FunctionalTests/QuicListenerTests.cs

```diff
@@ -87,6 +91,210 @@ public async Task AcceptConnectionAsync_ThrowingOptionsCallback_Throws(bool useF
+            {
+                connectAttempted.SetResult();
+                await serverDisposed.Task;
+                Assert.True(cancellationToken.IsCancellationRequested);
+                return null;
+            };
+            QuicListener listener = await CreateQuicListener(listenerOptions);
+
+            // One accept that will have an incoming connection from the client.
+            ValueTask<QuicConnection> acceptTask1 = listener.AcceptConnectionAsync();
+            // Another accept without any incoming connection.
+            ValueTask<QuicConnection> acceptTask2 = listener.AcceptConnectionAsync();
+
+            // Attempt to connect the first accept.
+            ValueTask<QuicConnection> connectTask = CreateQuicConnection(listener.LocalEndPoint);
+
+            // First, wait for the connect attempt to reach the server; otherwise, the client exception might end up being HostUnreachable.
+            // Then, dispose the listener and un-block the waiting server options callback.
+            await connectAttempted.Task;
+            await listener.DisposeAsync();
+            serverDisposed.SetResult();
+
+            var accept1Exception = await AssertThrowsQuicExceptionAsync(QuicError.OperationAborted, async () => await acceptTask1);
+            var accept2Exception = await AssertThrowsQuicExceptionAsync(QuicError.OperationAborted, async () => await acceptTask2);
+
+            Assert.Equal(accept1Exception, accept2Exception);
+
+            // Connect attempt should be stopped with "UserCanceled".
+            var connectException = await Assert.ThrowsAsync<AuthenticationException>(async () => await connectTask);
+            Assert.Contains(TlsAlertMessage.UserCanceled.ToString(), connectException.Message);
+        }
+
+        [Fact]
+        public async Task Listener_BacklogLimitRefusesConnection_ClientThrows()
+        {
+            QuicListenerOptions listenerOptions = CreateQuicListenerOptions();
+            listenerOptions.ListenBacklog = 2;
+            await using QuicListener listener = await CreateQuicListener(listenerOptions);
+
+            // The third connection attempt fails with ConnectionRefused.
+            await using var clientConnection1 = await CreateQuicConnection(listener.LocalEndPoint);
+            await using var clientConnection2 = await CreateQuicConnection(listener.LocalEndPoint);
+            await AssertThrowsQuicExceptionAsync(QuicError.ConnectionRefused, async () => await CreateQuicConnection(listener.LocalEndPoint));
+
+            // Accept one connection and attempt another one.
+            await using var serverConnection = await listener.AcceptConnectionAsync();
+            await using var clientConnection3 = await CreateQuicConnection(listener.LocalEndPoint);
+            // Third one again, should fail.
+            await AssertThrowsQuicExceptionAsync(QuicError.ConnectionRefused, async () => await CreateQuicConnection(listener.LocalEndPoint));
```

Comment:
we may check that `clientConnection2` can still be accepted. e.g. failure of `clientConnection3` does not impact `clientConnection3`.

Reply:
Added.